modules/ding_wayf/ding_wayf.module

```diff
@@ -124,6 +124,81 @@ function ding_wayf_init() {
   }
 }
 
+
+/**
+ * Do a check on the checksum given from gatewayf.
+ *
+ * @param array $attributes
+ *   The attributes from gatewayf.
+ *
+ * @return bool
+ *   Safe or not.
+ */
+function ding_wayf_security_check(array $attributes) {
+  // Check attributes.
+  if (empty($attributes['schacPersonalUniqueID']) || empty($attributes['eduPersonTargetedID'])) {
+    // We expect alway to get these two attributes.
+    watchdog('ding_gatewayf', 'Neccesary attributes missing', array(), WATCHDOG_ERROR);
+    drupal_set_message('Sorry login failed - contact site administrator', 'warning');
+    return FALSE;
+  }
+
+  $gateway_checksum = isset($attributes['hash'][0]) ? $attributes['hash'][0] : NULL;
+  if (!isset($gateway_checksum)) {
+    // This is an error from gatewayf - checksum should always be in attributes.
+    watchdog('ding_gatewayf', 'No checksum from gatewayf', array(), WATCHDOG_ERROR);
+    return FALSE;
+  }
+
+  $fixed_params = array(
+    $attributes['schacPersonalUniqueID'][0],
+    $attributes['eduPersonTargetedID'][0],
+  );
+
+  $key = ding_wayf_key_from_attributes($fixed_params);
+  // Get salt from settings $conf.
+  $salt = variable_get('ding_gatewayf_salt', NULL);
+  if (empty($salt)) {
+    // This is a configuration error.
+    watchdog('ding_gatewayf', 'No salt has been defined - wayf login is disabled for security', array(), WATCHDOG_ERROR);
+    return FALSE;
+  }
+  $ding_checksum = md5($key . $salt);
+  if ($ding_checksum !== $gateway_checksum) {
+    // Somebody has changed something - abort.
+    watchdog('ding_gatewayf', 'checksum from gatewayf did not match', array(), WATCHDOG_ERROR);
+    return FALSE;
+  }
+
+  return TRUE;
+}
+
+/**
+ * Make a salt based on request parameters.
```

Comment:
This docblock is somewhat misleading. This function does not generate a [salt](https://en.wikipedia.org/wiki/Salt_(cryptography)). The salt is the value set as a variable in `settings.php`.